### PR TITLE
Fixes #19963 - undefined method `string' during virt-who upload

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

**Comment by dLobatog** [comment]
[test] , not sure why this failed.

**Comment by jlsherrill** [comment]
In development .string() works fine, there is some difference when running with puma in development vs with passenger.

**Comment by jlsherrill** [comment]
Test failure seems unrelated, investigating .

File: app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb [hunk 1]
```diff
@@ -114,7 +114,7 @@ def rhsm_index
     # Note that this request comes in as content-type 'text/plain' so that
     # tomcat won't parse the json.  Here we just pass the plain body through to candlepin
     def async_hypervisors_update
-      raw_json = request.body.string
+      raw_json = request.body.read
```
Comment (ehelms) [inline app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb]: In a similar situation, the following was used: https://github.com/Katello/katello/blob/master/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb#L16

**Comment by sean797** [comment]
I guess that works as well @ehelms. Updated and Jenkins is doing it thing...

**Comment by jlsherrill** [comment]
ACK thanks @sean797 !

**Pull request merged**